src/OpenTelemetry.ResourceDetectors.Container/KubernetesContainerInfoFetcher.cs

```diff
@@ -0,0 +1,227 @@
+    {
+        this.apiConnector = apiConnector;
+        this.containerName = containerName;
+    }
+
+    public string ExtractContainerId()
+    {
+        // executing request only once. Do we need to retry again if data not available?
+        string response = this.ExecuteApiRequest();
+        if (response == null)
+        {
+            return string.Empty;
+        }
+
+        return this.ParseResponse(response);
+    }
+
+    internal static KubernetesContainerInfoFetcher? GetInstance()
+    {
+        bool isRequirementsPresent = CheckAndInitRequirements(out var apiConnector, out var containerName);
+
+        if (isRequirementsPresent && containerName != null)
+        {
+            return new KubernetesContainerInfoFetcher(apiConnector, containerName);
+        }
+
+        return null;
+    }
+
+    protected static bool CheckAndInitProp(string envPropName1, string? envPropName2, out string? result, bool canContinue = false)
+    {
+        string? value = Environment.GetEnvironmentVariable(envPropName1);
+
+        if (value == null && envPropName2 != null)
+        {
+            value = Environment.GetEnvironmentVariable(envPropName2);
+        }
+
+        if (value == null || string.IsNullOrEmpty(value))
+        {
+            result = null;
+
+            return false;
+        }
+
+        result = value;
+        return true;
+    }
+
+    protected static bool CheckFileAndInitProp(string dirName, string fileName, bool isCertFile, out string result)
+    {
+        result = string.Empty;
+        string filePath = Path.Combine(dirName, fileName);
+        try
+        {
+            FileInfo fileInfo = new(filePath);
+
+            if (!IsFilePresent(fileInfo))
+            {
+                return false;
+            }
+
+            // if this is certificate file, we don't have to read it yet but only check for existence and readability
+            // ca.cert will be directly consumed as input stream when building SSL context
+            if (isCertFile)
+            {
+                result = Path.GetFullPath(filePath);
+            }
+            else
+            {
+                string data = File.ReadAllText(filePath).Trim();
+
+                // file only has whitespaces
+                if (string.IsNullOrEmpty(data))
+                {
+                    return false;
+                }
+
+                result = data;
+            }
+        }
+        catch (Exception e)
+        {
+            ContainerExtensionsEventSource.Log.ExtractResourceAttributesException("Cannot Read " + Path.GetFullPath(filePath) + " : " + e.Message, e);
+            return false;
+        }
+
+        return true;
+    }
+
+    protected static string FormatContainerId(string unFormattedId)
+    {
+        // "containerID"="docker://18e1f4b72f6861b5e591e11ea6db0640377de6ed5dc9bffbae4d9ab284d53044"
+        // Assuming kube api return container id always in this format prefixed with 'docker://'. (Big assumption?)
```

Comment:
The id can start with `containerd://` as well, and probably with something else in the case of another container runtime.

Reply:
Probably we should update comment. Code as written will take everything after last "/" and will validate that it looks like container id (hex string)